### PR TITLE
Codesplitting & heroku friendlyfied.

### PR DESCRIPTION
**Pull request by antonedvard**
So, here we go.

#### Added a .env.example file
Within the env. variables i changed `P4NT3RB0T-AUTH` to simply `BOT_AUTH` and added the PORT env.

#### .gitignore
Swapped out our homemade .gitignore with the node.js standard .gitignore from https://github.com/antonedvard/gitignore/blob/master/Node.gitignore

#### package.json
Moved the `@types` dependencies from `dev-dependencies` to `dependencies`, that seemed to fix the building within the heroku (The error where the guy wanted us to create a procfile)

#### procfile
obviously that shitty file was not needed so it was removed...

#### server.ts
removed because of the code splitting, the entry file became `app.ts`

#### tsconfig.json
removed `"types": ["express", "tmi.js"],` by some unknown reason the build in heroku threw an error when we included that line...

#### rest of the .ts files
created from the code splitting, done to make the code a little bit more readable and understandable.

My pb-mainframe-dev build and run without issues on heroku, you have access to it there.
Happy days! :)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,118 @@
...
+*.lcov
+
+# nyc test coverage
+.nyc_output
+
+# Grunt intermediate storage (https://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
+
+# Bower dependency directory (https://bower.io/)
+bower_components
```
Comment (whitep4nth3r) [inline .gitignore:32]: Do we really need _all_ of this? I haven't used bower in FOREVER!

File: src/app.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { config } from "dotenv";
+config();
+
+import { webServer } from "./webserver";
+import { tmi } from "./tmi";
+
+async function run() {
```
Comment (whitep4nth3r) [inline src/app.ts:7]: This splitting out is excellent

File: .gitignore (hunk 1 above)
Comment (antonedvard) [reply .gitignore:32]: Well i guess its just for good measures, if others fork and use their own its allright to have it lying around, ofcourse it is your call :)

**Comment by whitep4nth3r** [comment]
@antonedvard This looks so much simpler! Thank you so much! There are some conflicts in package.json and tsconfig - I'm sure you can just 'take yours' in all of the conflicts but would you might merging in the existing master to sort out the conflicts? Then we will merge!

**Comment by whitep4nth3r** [comment]
@all-contributors please add @antonedvard for Code

**Comment by whitep4nth3r** [comment]
@all-contributors please add @antonedvard for Code

File: .gitignore (hunk 1 above)
Comment (whitep4nth3r) [reply .gitignore:32]: That's a great point!

**Pull request merged**